### PR TITLE
Make PUT operation on portalsettings resources ARM-compliant

### PR DESCRIPTION
**Pull request by promoisha**

### CODE REVIEW

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutSignUp.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          "text": "Terms of service text.",
+          "consentRequired": true
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+          "properties": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutSignUp.json]: doesnt it have the "id", "name", "type"?

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutSignIn.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "parameters": {
+      "properties": {
+        "enabled": true
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+          "properties": {
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutSignIn.json]: "id", "name", "type" across all examples

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutDelegation.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "parameters": {
+    "serviceName": "apimService1",
+    "resourceGroupName": "rg1",
+    "api-version": "2018-01-01",
+    "subscriptionId": "subid",
+    "uid": "5931a75ae4bbd512288c680b",
+    "If-Match": "*",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutDelegation.json:8]: > "If-Match": "*", [](start = 3, length = 17)

is if-Match a must

Comment (promoisha) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-01-01/examples/ApiManagementPortalSettingsPutDelegation.json:8]: yup, as we take into account the "version" of all the settings under portalsettings

**Comment by promoisha** [comment]
@solankisamir Updated the PR

**Pull request merged**